pkg/actors/actors.go

```diff
@@ -1062,30 +1063,46 @@ func (a *actorsRuntime) CreateTimer(ctx context.Context, req *CreateTimerRequest
 		close(stopChan.(chan bool))
 	}
 
-	if period, err = time.ParseDuration(req.Period); err != nil {
-		return err
+	repeats = -1 // set to default
+	if len(req.Period) != 0 {
+		if period, repeats, err = parseDuration(req.Period); err != nil {
+			return err
+		}
 	}
+
+	// error on timers with zero repetitions
+	if repeats == 0 {
+		return errors.Errorf("timer %s has zero repetitions", timerKey)
+	}
+
 	if len(req.DueTime) > 0 {
-		if dueTime, err = time.ParseDuration(req.DueTime); err != nil {
+		if dueTime, _, err = parseDuration(req.DueTime); err != nil {
 			return err
 		}
 	}
 
+	if len(req.TTL) > 0 {
+		if ttl, _, err = parseDuration(req.TTL); err != nil {
+			return err
+		}
+		// error on expired timers
+		now := time.Now()
+		if now.After(now.Add(ttl)) {
+			return errors.Errorf("timer %s has expired ttl %s", timerKey, ttl.String())
+		}
+	}
+
 	stop := make(chan bool, 1)
 	a.activeTimers.Store(timerKey, stop)
 
 	go func(stop chan bool, req *CreateTimerRequest) {
 		// Check if timer is still active
-		timer := time.NewTimer(dueTime)
 		select {
 		case <-time.After(dueTime):
 			log.Debugf("Time: %v with parameters: DueTime: %v, Period: %v, Data: %v has been overdue.", timerKey, req.DueTime, req.Period, req.Data)
 			break
 		case <-stop:
 			log.Infof("Time: %v with parameters: DueTime: %v, Period: %v, Data: %v has been deleted.", timerKey, req.DueTime, req.Period, req.Data)
-			// Stop timer to free resource
-			timer.Stop()
-			timer = nil
 			return
 		}
 
```

Comment:
As discussed we need to do the TTL duration parsing here so that our TTL is computed as a duration after the dueTime has passed (if provided):

```
	if len(req.TTL) > 0 {
		if ttl, _, err = parseDuration(req.TTL); err != nil {
			return err
```

Reply:
good catch, thank you for reviewing!